scripts/mbedtls_dev/bignum_mod_raw.py

```diff
@@ -51,6 +51,37 @@ def result(self) -> List[str]:
         result = (self.int_a - self.int_b) % self.int_n
         return [self.format_result(result)]
 
+class BignumModRawFixQuasiReduction(bignum_common.ModOperationCommon,
+                                    BignumModRawTarget):
+    """Test cases for ecp quasi_reduction()."""
+    symbol = "-"
+    test_function = "mpi_mod_raw_fix_quasi_reduction"
+    test_name = "mbedtls_mpi_mod_raw_fix_quasi_reduction"
```

Comment:
That should generate a generated.data file 

~~~~~~
mbedtls_mpi_mod_raw_fix_quasi_reduction and some description
mpi_mod_raw_fix_quasi_reduction:param_N:param_x:param_result
~~~~~~

I feel that it should be setting up a proper dependecy

~~~~~~~
mbedtls_mpi_mod_raw_fix_quasi_reduction and some description
depends_on:MBEDTLS_TEST_HOOKS
mpi_mod_raw_fix_quasi_reduction:param_N:param_x:param_result
~~~~~~~

I think setting the object's dependencies would properly set that

~~~~~~
dependencies = ["MBEDTLS_TEST_HOOKS"]       # If called as a class object
self.dependencies = ["MBEDTLS_TEST_HOOKS"]  # If called from a method
~~~~~~

Reply:
I haven't looked at the code but this review comment looks weird. I would not expect to find `MBEDTLS_TEST_HOOKS` in a `.data` file, only in a `.function` file, since it's about the availability of auxiliary functions. How would you have test data that requires `MBEDTLS_TEST_HOOKS`, as opposed to the function `mpi_mod_raw_fix_quasi_reduction` requiring `MBEDTLS_TEST_HOOKS`?

Reply:
In fact I'd go so far as to say: the presence of `MBEDTLS_TEST_HOOKS` in a `.data` file would indicate that something is wrong. `MBEDTLS_TEST_HOOKS` may be necessary because some test code doesn't build otherwise (because it's testing an internal function, or because it's relying on some hooks to inspect internal data structures). `MBEDTLS_TEST_HOOKS` should never affect the behavior of a library function, so if it makes a test pass, it's being used wrong.

Reply:
The main reason for the comment is that the test `.function` [depends on it](https://github.com/Mbed-TLS/mbedtls/blob/1e8c210b9d1f370c9724edbd7d86089fb4b58c68/tests/suites/test_suite_bignum_mod_raw.function#L343), and following what the [documentation ](https://github.com/Mbed-TLS/mbedtls/blob/development/docs/architecture/testing/invasive-testing.md)instructs for that  type of testing.

I could always be misinterepting the instructions or the intent of this PR's changes.

Reply:
Yes, sure, the unit test **function** that's testing an otherwise-static function depends on `MBEDTLS_TEST_HOOKS`. Why do you want to add a dependency in the **data** file?

Reply:
I too think that this dependency should be in .function file.